### PR TITLE
In case of error when starting sanic

### PR DESCRIPTION
**Pull request by ticosax**
Don't exit with code 0

### CODE REVIEW

File: sanic/app.py [hunk 1]
```diff
@@ -570,9 +570,10 @@ def run(self, host="127.0.0.1", port=8000, debug=False, before_start=None,
                 serve(**server_settings)
             else:
                 serve_multiple(server_settings, workers)
-        except:
+        except Exception:
```
Comment (r0fls) [inline sanic/app.py]: this behaves the same as a bear `except:`

Comment (ticosax) [inline sanic/app.py]: Not really, a bare `except:` captured everything that also inherits `BaseException` including signals like KeyboardInterrupt or SystemExit.
that you probably don't want to capture here. it is considered anti-pattern.
Your call though.

Comment (seemethere) [inline sanic/app.py]: We do want it to catch `KeyboardInterrupt` though

Comment (ticosax) [inline sanic/app.py]: > We do want it to catch KeyboardInterrupt though

Fair enough, I will revert that change

Comment (messense) [inline sanic/app.py]: I would think if you want to catch everything, using

```python
try:
    ...
except BaseException:
    ...
```

is better than bare `except:` as it is much more clear.

**Pull request merged**